llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -5805,6 +5806,10 @@ Register AMDGPULegalizerInfo::fixStoreSourceType(
   if (hasBufferRsrcWorkaround(Ty))
     return castBufferRsrcToV4I32(VData, B);
 
+  if (shouldBitcastLoadStoreType(ST, Ty, MemTy) || Ty.isPointerVector()) {
```

Comment:
I'm not sure why you need to touch any of the globalisel code, but this should be a separate change?

Reply:
I found crashes in global isel while working on this patch, but yeah, this should be a separate PR now that you mention it